### PR TITLE
Add MethodOfMomentsFactory

### PR DESCRIPTION
**Pull request by jschueller**
Allows to estimate a parametric model from moments in a generic manner.

### CODE REVIEW

File: lib/src/Uncertainty/Distribution/MethodOfMomentsFactory.cxx [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+    }
+
+    // compute sum of deltas between centered homogenized moments
+    NumericalScalar result = 0.0;
+    for (UnsignedInteger j = 0; j < parameterDimension; ++ j)
+    {
+      const NumericalScalar signref = refMoments_[j] > 0.0 ? 1.0 : -1.0;
+      const NumericalScalar sign = moments[j] > 0.0 ? 1.0 : -1.0;
+      result += std::abs(signref * std::pow(std::abs(refMoments_[j]), 1.0 / (j + 1.0)) - sign * std::pow(std::abs(moments[j]), 1.0 / (j + 1.0)));
+    }
```
Comment (regislebrun) [inline lib/src/Uncertainty/Distribution/MethodOfMomentsFactory.cxx]: Here you have to compare the normalized moments, ie you have to compute:
[ |\mu_1-m_1|/\sigma ]^2 + \sum_{i=2}^K[ (\epsilon_i|\mu_i|^{1/i}-\eta_i|m_i|^{1/i})/\sigma ]^2
where \mu_1 is the mean of the distribution, m_1 the mean of the sample, \mu_i the centered moment of order i>=2 of the distribution, m_i the associated sample centered moment and \sigma is either the sample standard deviation or the distribution standard deviation. Note that you must use the same value to normalize the distribution moments and the sample moments. If you use the distribution standard deviation to normalize the distribution centered moments and the sample standard deviation to normalize the sample centered moments, the first term in the sum is equal to 0 and you loose one moment equation!

File: lib/test/t_Axial_std.cxx [hunk 2]
```diff
@@ -46,7 +46,7 @@ int main(int argc, char *argv[])
     Axial myPlane(center, levels);
     fullprint << "myPlane = " << myPlane << std::endl;
     NumericalSample sample(myPlane.generate());
-    fullprint << "sample = " << sample << std::endl;
+    fullprint << "sample = " << std::pow(-0.8, 1.0/3) << std::endl;
   }
```
Comment (regislebrun) [inline lib/test/t_Axial_std.cxx]: ???

File: python/src/MethodOfMomentsFactory_doc.i.in [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+Notes
+-----
+The underlying optimization problem seeks to minimize the sum of slacks between the
+centered moments of the sample and the centered moments of the parametric model:
+
+.. math::
+
+    \Delta = \argmin_{\vect{\theta} \in \Rset^P} \sum_{k=1}^P |(\widehat{\vect{m}}^{(k)}_0)^{\frac{1}{k}} - (\vect{m}^{(k)}_0)^{\frac{1}{k}}|
+
```
Comment (regislebrun) [inline python/src/MethodOfMomentsFactory_doc.i.in]: Update the description using my previous comment

File: lib/src/Uncertainty/Distribution/MethodOfMomentsFactory.cxx (hunk 1 above)
Comment (jschueller) [inline lib/src/Uncertainty/Distribution/MethodOfMomentsFactory.cxx]: it helps a lot!

**Comment by regislebrun** [comment]
+1

File: lib/src/Uncertainty/Distribution/MethodOfMomentsFactory.cxx [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+    for (UnsignedInteger j = 1; j < parameterDimension; ++ j)
+    {
+      moments[j] = distribution.getCenteredMoment(j + 1)[0];
+    }
+
+    // compute sum of deltas between centered homogenized moments
+    NumericalScalar result = 0.0;
+    for (UnsignedInteger j = 0; j < parameterDimension; ++ j)
+    {
+      const NumericalScalar signRef = refMoments_[j] > 0.0 ? 1.0 : -1.0;
```
Comment (sofianehaddad) [inline lib/src/Uncertainty/Distribution/MethodOfMomentsFactory.cxx]: we could also define NumericalPoint signRef_ et set once the signs

File: python/src/MethodOfMomentsFactory_doc.i.in [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+Notes
+-----
+The underlying optimization problem seeks to minimize the sum of slacks between the
+centered moments of the sample and the centered moments of the parametric model:
+
+.. math::
+
+    \Delta = \argmin_{\vect{\theta} \in \Rset^K} [ |\mu_1-m_1|/\sigma ]^2 + \sum_{i=2}^K[ (\epsilon_i|\mu_i|^{1/i}-\eta_i|m_i|^{1/i})/\sigma ]^2
+
```
Comment (sofianehaddad) [inline python/src/MethodOfMomentsFactory_doc.i.in]: definition of mu_i, m_i, epsilon_i, eta_i ?

Comment (jschueller) [inline python/src/MethodOfMomentsFactory_doc.i.in]: added mu_i, m_i, and removed epsilon, eta ; it's better I do not explicit them

**Pull request merged**